File: src/courses/data/catalog/02-dsa/dsa-core/implementations/c/dynamic-programming.c
/*
 * C Implementation: Dynamic Programming
 * Invariant: Optimal Substructure + Overlapping Subproblems
 */
#include <stdio.h>
#include <stdlib.h>

typedef struct {
    int* data;
    int size;
    int capacity;
} dynamic_programming_t;

// TODO: Complete the allocation and operations
void init_dynamic_programming(dynamic_programming_t* item, int capacity) {
    item->data = (int*)malloc(capacity * sizeof(int));
    item->size = 0;
    item->capacity = capacity;
}

File: src/courses/data/catalog/02-dsa/dsa-core/implementations/java/dynamic-programming.java
/**
 * Java Implementation: Dynamic Programming
 * Invariant: Optimal Substructure + Overlapping Subproblems
 */
package seed.dsa.core;

import java.util.*;

public class DynamicProgramming<T> {
    private final List<T> items = new ArrayList<>();

    // TODO: Implement your solution here
    public void add(T value) {
        items.add(value);
    }

    public int size() {
        return items.size();
    }
}
